### PR TITLE
fix: update correct path to use when exporting 404 page

### LINKED ISSUE #5035: Export 404.html when running `next export` without `exportPathMap`
# Feature request

## Is your feature request related to a problem? Please describe.

`next export`'s `defaultPathMap` contains `/_error` which is not supported as error page by most browsers. What however is supported is `404.html`.

## Describe the solution you'd like

Updating https://github.com/zeit/next.js/blob/canary/server/export.js#L34

To have the following statement:

```
if(page === '/_error') {
  defaultPathMap['404'] = { page }
  continue
}
```

This will make `next export` export `404.html` with the contents of `/_error`

## Describe alternatives you've considered

The alternative is doing this manually in `exportPathMap`

```
module.exports = {
  exportPathMap(defaultPathMap) {
    defaultPathMap['404'] = defaultPathMap['/_error']
    return defaultPathMap
  }
}
```

## Additional context

Feature request originated on this spectrum.chat thread: https://spectrum.chat/thread/f765b794-9aa9-4edc-b549-6de2268847e5

### PR DESCRIPTION
**Pull request by CorentinAndre**
When exporting error page, next defaults it to 404/index.html which is not recognized as a default 404 page.

Fixes https://github.com/zeit/next.js/issues/5035

### CODE REVIEW

**Comment by timneutkens** [comment]
Actually this:

```
defaultPathMap['/404'] = { page }
```

can be changed to this:

```
defaultPathMap['/404.html'] = { page }
```

And it will be exported as such, as it will go into the first if statement (if there's an extension)

File: packages/next/export/index.js [hunk 1]
```diff
@@ -128,6 +128,9 @@ export default async function (dir, options, configuration) {
     } else if (path === '/') {
       // If the path is the root, just use index.html
       htmlFilename = 'index.html'
+    }else if (path === '/404') {
```
Comment (timneutkens) [inline packages/next/export/index.js]: This can be removed.

**Comment by timneutkens** [comment]
Could you add a test for this to `test/integration/export` 👍

File: packages/next/export/index.js (hunk 1 above)
Comment (CorentinAndre) [reply packages/next/export/index.js]: My bad

**Comment by kylemh** [comment]
Thanks for getting this! Haven't had a moment to do it.

**Comment by timneutkens** [comment]
I've fixed the code and added a test 👌

**Review by shuding — APPROVED** [review]

**Comment by timneutkens** [comment]
Thanks @CorentinAndre!

**Pull request merged**